web-common/src/main/java/org/jboss/as/web/session/CompositeCharSequence.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.web.session;
+
+import java.nio.CharBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A {@link CharSequence} composed of other character sequences.
+ * All methods delegate to one or more of the underlying character sequences, using relative indexes.
+ * @author Paul Ferraro
+ */
+public class CompositeCharSequence implements CharSequence {
```

Comment:
This can be package protected.

Reply:
Good point.  It's a generally reusable class, however, this isn't the appropriate module for such a public class.  I'll make this package protected and submit a PR to add this to wildfly-common (seems to be an appropriate dumping ground) and drop this class once a version containing it is integrated.